### PR TITLE
[compiler-rt] Make add_custom_libcxx() resilient to DESTDIR being set

### PR DESCRIPTION
**Pull request by arichardson**
If DESTDIR is set in the environment during the build/test stage, the
local libc++ installation will be installed under DESTDIR instead of being
in the build directory.

See https://github.com/llvm/llvm-project/pull/115077#issuecomment-2464640457
and https://gitlab.kitware.com/cmake/cmake/-/issues/18165.

### CODE REVIEW

File: compiler-rt/cmake/Modules/AddCompilerRT.cmake [hunk 1]
```diff
@@ -713,6 +713,9 @@ macro(add_custom_libcxx name prefix)
     USES_TERMINAL_INSTALL 1
     LIST_SEPARATOR |
     EXCLUDE_FROM_ALL TRUE
+    # Ensure that DESTDIR=... set in the out environment does not affect this
+    # target (we always need to install to the build directory).
+    INSTALL_COMMAND env DESTDIR= ${CMAKE_COMMAND} --build ${prefix}/build --target install
     INSTALL_BYPRODUCTS "${prefix}/lib/libc++.a" "${prefix}/lib/libc++abi.a"
```
Comment (bjope) [inline compiler-rt/cmake/Modules/AddCompilerRT.cmake:719]: With the INSTALL_COMMAND fix we now get this:
```
[2091/2122] Performing install step for 'libcxx_fuzzer_x86_64'
ninja: error: unknown target 'INSTALL_BYPRODUCTS'
FAILED: compiler-rt/lib/fuzzer/libcxx_fuzzer_x86_64/src/libcxx_fuzzer_x86_64-stamp/libcxx_fuzzer_x86_64-install 
```

Comment (bjope) [reply compiler-rt/cmake/Modules/AddCompilerRT.cmake:719]: Ah, maybe this patch bumps the requirement on cmake version to 3.26 (while current requirement is 3.20.0 according to https://llvm.org/docs/GettingStarted.html#software).

Comment (arichardson) [reply compiler-rt/cmake/Modules/AddCompilerRT.cmake:719]: Ah I guess I had a newer CMake, we can make this conditional on CMake version.

Comment (arichardson) [reply compiler-rt/cmake/Modules/AddCompilerRT.cmake:719]: I just built with CMake 3.20 and am able to reproduce - will fix now.

Comment (arichardson) [reply compiler-rt/cmake/Modules/AddCompilerRT.cmake:719]: Fixed by https://github.com/llvm/llvm-project/pull/115677

**Pull request merged**